PerimeterXModule/PxModule.cs

```diff
@@ -317,21 +320,48 @@ public static void ResponseBlockPage(PxContext pxContext, PxModuleConfigurationS
             {
                 template = "challenge";
             }
-            else if (pxContext.BlockAction == "c")
-            {
-                template = config.CaptchaProvider;
-            }
-
-            Debug.WriteLine(string.Format("Using {0} template", template), PxConstants.LOG_CATEGORY);
-
-            // In the case of a challenge, the challenge response is taken directly from BlockData. Otherwise, generate html template.
-            content = template == "challenge" && !string.IsNullOrEmpty(pxContext.BlockData) ? pxContext.BlockData :
-                TemplateFactory.getTemplate(template, config, pxContext.UUID, pxContext.Vid);
-
-            pxContext.ApplicationContext.Response.Write(content);
-        }
-
-        public void Dispose()
+			else if (pxContext.BlockAction == "c")
+			{
+				template = config.CaptchaProvider;
+			}
+
+			if (pxContext.CookieOrigin.Equals(CookieOrigin.HEADER))
+			{
+				template = string.Format("{0}Mobile", template);
+			}
+			Debug.WriteLine(string.Format("Using {0} template", template), PxConstants.LOG_CATEGORY);
+
+			// In the case of a challenge, the challenge response is taken directly from BlockData. Otherwise, generate html template.
+			string html = template == "challenge" && !string.IsNullOrEmpty(pxContext.BlockData) ? pxContext.BlockData :
+				TemplateFactory.getTemplate(template, config, pxContext.UUID, pxContext.Vid);
+			// set html as default content for mobile scenario
+			content = html;
+			pxContext.ApplicationContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
+			if (pxContext.CookieOrigin.Equals(CookieOrigin.HEADER))
+			{
+				pxContext.ApplicationContext.Response.ContentType = "application/json";
+				using (var output = new StringWriter())
+				{
+					JSON.Serialize(
+						new MobileResponse()
+						{
+							AppId = config.AppId,
+							Uuid = pxContext.UUID,
+							Action = pxContext.MapBlockAction(),
+							Vid = pxContext.Vid,
+							Page = Convert.ToBase64String(Encoding.UTF8.GetBytes(html)),
+							CollectorUrl = string.Format(config.CollectorUrl, config.AppId)
+						},
+
+						output
```

Comment:
Please remove the empty lines. I'd actually place the `output` after the comma - 
```
...
}, output);
```